### ISSUE TITLE
Add a list of functions

### ISSUE THREAD

**Issue by tpoisot** [datacarpentry/sql-ecology-lesson#75]
In `01-...`, the "Calculated values" header mentions functions. Maybe a table would be nice.

**Comment by waveform80** [comment]
Good idea! Some functions common to all databases (that I'm aware of), which are probably worth including:

* ABS
* COALESCE (sometimes aliased to NVL, IFNULL, or VALUE but COALESCE is the standard name, supported by all engines)
* LENGTH
* LOWER
* NULLIF
* ROUND
* TRIM (most engines also support LTRIM and RTRIM)
* UPPER

Some functions which appear in every engine but have minor variations

* GREATEST/LEAST (sometimes implemented as multi-valued MIN/MAX - SQLite is one of the latter)
* RANDOM (some engines like SQLite produce an integer, some a float between 0 and 1 - I *think* the float variant is the more common)
* REPLACE
* SUBSTRING (frequently SUBSTR, and many engines don't support the standard's weird "from ... for" syntax)

And then there's the debacle of datetimes that no engine can agree on (either in terms of functions or even the data-types). Not sure whether it's worth tackling this for the favoured engine (SQLite) or just leaving it out altogether?

**Comment by tpoisot** [comment]
As my first action as the maintainer of this lesson, I thereby assign ­you @waveform80 to this issue.

You can probably wait until #76 is merged, though.

I think once we have the basis covered, we can re-open an issue to discuss time/date. This is important for 99% of ecological data, so we should probably discuss the best strategy to mention it in the lessons.

**Comment by waveform80** [comment]
Sounds good to me :) I'll see what I can get done at the weekend (no worries about waiting on other commits - I can always rebase on master to keep the PR simple and merge-able).

**Comment by waveform80** [comment]
Finally got around to making a PR for this (#82). Some observations / questions:

* I've tried to keep it relatively short (similar length to other sections) - too short? Too long?

* Is it in the right place? I've added it straight after joins given the pertinence of COALESCE to this facility (I've not detailed COALESCE's role in OUTER JOINs as those aren't covered in the lesson). Should it be in its own section or is this place fine?

* Too much emphasis on COALESCE / NULLIF? Personally I think they're the most important functions as they're useful in numerous clauses, and they're also likely to be functions that users aren't familiar with (one can find analogs of ABS, LENGTH, SUBSTR, TRIM, UPPER, etc. in almost any language but COALESCE and NULLIF are quite peculiar to SQL).

* Are the challenges reasonable? I don't think they require any skills not already covered in earlier sections.

Finally, I tested rendering locally with jekyll and the tables look like they could use a little tweaking style-wise (more bottom padding and perhaps some header tweaks), but I'm not sure I'm building it exactly the same as the main site (no Makefile in this repo?).
